src/modules/Sticky/Sticky.js

```diff
@@ -0,0 +1,230 @@
+    onBottom: PropTypes.func,
+
+    /**
+     * Callback when element is fixed to page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onStick: PropTypes.func,
+
+    /* Callback when element is bound to top of parent container. */
+    onTop: PropTypes.func,
+
+    /**
+     * Callback when element is unfixed from page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     * */
+    onUnstick: PropTypes.func,
+  }
+
+  static defaultProps = {
+    offset: 0,
+    bottomOffset: 0,
+  }
+
+  static _meta = {
+    name: 'Sticky',
+    type: META.TYPES.MODULE,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = { sticky: false, top: null, bottom: null }
+  }
+
+  componentDidMount() {
+    this.update()
+    window.addEventListener('scroll', this.update)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.update)
+  }
+
+  calcBoundingRects() {
+    this.triggerBoundingRect = this.triggerRef.getBoundingClientRect()
+    this.contextBoundingRect = (this.props.context || document.body).getBoundingClientRect()
+    this.stickyBoundingRect = this.stickyRef.getBoundingClientRect()
+  }
+
+  // Return true when the component reached the bottom of the context
+  hasReachedContextBottom() {
+    return this.stickyBoundingRect.height + this.props.offset >= this.contextBoundingRect.bottom
+  }
+
+  // Return true when the component reached the starting point
+  hasReachedStartingPoint() {
+    return this.stickyBoundingRect.top <= this.triggerBoundingRect.top
+  }
+
+  // Return true when the top of the screen overpasses the Sticky component
+  hasTouchedScreenTop() {
+    return this.triggerBoundingRect.top < this.props.offset
+  }
+
+  // Return true when the bottom of the screen overpasses the Sticky component
+  hasTouchedScreenBottom() {
+    return this.contextBoundingRect.bottom + this.props.bottomOffset > window.innerHeight
+  }
+
+  // Return true if the height of the component is higher than the window
+  isOversized() {
+    return this.stickyBoundingRect.height > window.innerHeight
+  }
+
+  // If true, the component will stick to the bottom of the screen instead of the top
+  setPushing(pushing) {
+    if (this.props.pushing) this.setState({ pushing })
+  }
+
+  stickToContextTop(e) {
+    if (e && this.props.onTop) this.props.onTop(e)
+    this.unstick(e)
+    this.setPushing(false)
+  }
+
+  stickToContextBottom(e) {
+    if (e && this.props.onBottom) this.props.onBottom(e)
+    this.stick(e)
+    this.setState({
+      top: this.contextBoundingRect.bottom - this.stickyBoundingRect.height,
+      bottom: null,
+    })
+    this.setPushing(true)
+  }
+
+  stickToScreenTop(e) {
+    this.stick(e)
+    this.setState({ top: this.props.offset, bottom: null })
+  }
+
+  stickToScreenBottom(e) {
```

Comment:
Arrow functions implicitly bind the method to `this` when the instance is created, right? Isn't this kind of bad for performance, since we duplicate the function for each instance?